### ISSUE TITLE
[Request] Compatibility with Firefox for Android

### ISSUE THREAD

**Issue by tonogram** [kagisearch/browser_extensions#3]
It would be nice if the Kagi Search for Firefox extension was available to Firefox users on Android, if such a thing is possible. Currently you can manually add Kagi as a search engine (Settings > Search > Add search engine) but this relies on your session persisting, and does not work in private tabs (requiring you to log in again). Certainly not the worst thing in the world, but it would be nice to have.

**Comment by vprelovac** [comment]
Do you know what is currently preventing the extension from working on Android?

**Comment by tonogram** [comment]
I have no idea, I've never touched browser extension development and know next to nothing about how it actually works. For all I know, getting the Kagi extension to work on mobile could simply be impossible at this time, or it could be as simple as checking a "works on mobile" tick box on the Firefox extensions site. All I know for certain is that the install button is disabled on mobile.

**Comment by ReimarPB** [comment]
I tried running the current Firefox extension on Firefox for Android, and it runs but for some reason it's not able to capture the kagi_session cookie when opening the Kagi website, meaning it does not work in private windows. I tried debugging it but haven't been able to find out why.

Similarly, I was not able to get it to work in private mode when manually pasting the token in the addon either. I made sure the extension was enabled for private browsing and the same token worked on desktop, but it still did not work.

**Comment by vprelovac** [comment]
@BrunoBernardino Can you figure out what is needed for extension to work on FF Android

**Comment by vprelovac** [comment]
I'v eenabled the flag for Android support in 0.3.2 release, not sure if that helps?

**Comment by BrunoBernardino** [comment]
I haven't investigated this yet, but we can certainly test once it's out!

**Comment by BrunoBernardino** [comment]
v0.3.3 still shows to me as incompatible with Firefox for Mobile.

**Comment by vprelovac** [comment]
> v0.3.3 still shows to me as incompatible with Firefox for Mobile.

I did enable the checkbox so there must be something else Firefox for Android is looking for.

**Comment by tonogram** [comment]
Per [addons.mozilla.org/.../firefox/...](https://addons.mozilla.org/en-US/firefox/addon/kagi-search-for-firefox/) on my Android phone, I still cannot install the extension. It may be worth noting that there's a separate page for Android-compatible addons ([addons.mozilla.org/.../android/...](https://addons.mozilla.org/en-US/android)) in which Kagi does not appear in any [search results](https://addons.mozilla.org/en-US/android/search/?q=kagi).

I assume *someone* has gotten it working on their own Android device since this has been closed, but as far as I can tell it's still not available through traditional means.

**Comment by BrunoBernardino** [comment]
@tonogram indeed we closed this as we had done everything the Firefox for Android documentation told us, and it's possible to install it via a convoluted process of adding a custom collection in Beta/Nightly versions, but it doesn't quite work, as noted in https://kagifeedback.org/d/1768-summarizer-mobile-extension

It's something that's very hard to develop for and not really well documented, so I'll give it a shot again in a week or so, but if someone else has tips, we'd appreciate it.

**Comment by tonogram** [comment]
Thanks @BrunoBernardino and friends, I appreciate the work you're doing. Fingers crossed that something can be done.

**Comment by BrunoBernardino** [comment]
As per #26 the extension will look better once that's merged, but Firefox for Android still doesn't support anything we do, here, unfortunately. We'll have to take a look at this again in a couple of months, to see if anything changed.

**Comment by vallode** [comment]
I did not see this article mentioned here, so I thought it's worth linking: https://blog.mozilla.org/addons/2023/08/10/prepare-your-firefox-desktop-extension-for-the-upcoming-android-release/

It mentions an early September release (clearly missed now) but it also mentions that some changes might be required on the background scripts.

**Comment by BrunoBernardino** [comment]
Thanks @vallode ! That article is mentioned in the linked PR right above your comment. Unfortunately it's still not working with that.

**Comment by brad-anderson** [comment]
Firefox Nightly for Android seems to have cracked open the extensions. I added Kagi just fine.

The Kagi extensions doesn't seem to do anything currently, though. It doesn't recognize when you are signed in or affect available search providers but it's a start.

This indirectly made it much easier for me to use Kagi on mobile because now I can use the Cookie Auto Delete extension rather than only using Private Tabs to get my clean browsing fix (though I recently noticed the new advice for using the Session Link to get Kagi on private tabs which would have worked too).

**Comment by runiq** [comment]
FYI, Firefox 120 will be when the floodgates open and lots of addons from AMO will be available on Android as well. https://blog.mozilla.org/addons/2023/11/01/is-your-extension-ready-for-firefox-for-android/

**Comment by BrunoBernardino** [comment]
Thanks @runiq I've been paying attention but so far there's nothing we need to change, as far as we can tell.

**Comment by UndarkAido** [comment]
The automatic search engine addition and universal summarizer shortcut don't work

**Comment by BrunoBernardino** [comment]
@tonogram can you please confirm that?

**Comment by tonogram** [comment]
I can corroborate that the Universal Summarizer does not work. The Summarize button seemingly does nothing.

> automatic search engine addition
I assume this means adding Kagi to the list of available default search engines. I had already added Kagi manually but ostensibly the extension does not add the option itself.

Apologies for jumping the gun, I was only thinking about preserving the session and not about the other features. I'll update the original issue to accurately reflect the current problems.

**Comment by tonogram** [comment]
@meatyalien Opening a new private session does work with Kagi, for me.

**Comment by BrunoBernardino** [comment]
@meatyalien can you please try, in the extension's permissions/settings, disabling the "Run in Private Windows" toggle, saving the token, then enabling it again?

**Comment by BrunoBernardino** [comment]
Great! That solution should show up inline if your extension fails to save the session token manually since v0.4.3.

**Comment by noodlejetski** [comment]
it doesn't look like disabling the "allow.in private mode" toggle, pasting and saving my generated token, and enabling the toggle again does the trick for me, I'm still asked to log in whenever I try searching in a private tab. using Firefox 121 on Android 13.

**Comment by noodlejetski** [comment]
@meatyalien yes, I've done that and am still greeted with a login screen in private mode. not sure how I'm supposed to "make sure it works in normal browsing mode" in step 6 since I've already logged in step 5, and there's not much else the addon does.

**Comment by BrunoBernardino** [comment]
0.6.0 was the latest one available, but Mozilla removed it because they demand a privacy consent screen now (because the summarizer, when used, sends the current active tab's URL to Kagi). It's being fixed in #78 and once the legal piece of the full extension privacy policy is finished, should be live, but @vprelovac should know more.

**Comment by andrewducker** [comment]
I am delighted that Firefox take privacy seriously, and that you're getting it sorted!

**Comment by BrunoBernardino** [comment]
To be honest, the extension doesn't send anything the users don't authorize explicitly (the permissions the extension requests are already very specific, narrow, and require user opt-in), so it felt surprising and a bit targeted, but that's just my personal opinion.

**Comment by pabpas** [comment]
Is there any workaround for incognito mode until the extension comes back?

**Comment by Hund** [comment]
Any updates on this issue? It would be nice to have a working search on Firefox for Android.

**Comment by PC-SPEZIALIST** [comment]
We would love to use the Kagi Firefox extension on Android too.
